components/script/script_thread.rs

```diff
@@ -155,7 +154,8 @@ struct InProgressLoad {
     is_visible: bool,
     /// The requested URL of the load.
     url: ServoUrl,
-    origin: Origin,
+    /// The origin for the document
+    origin: MutableOrigin,
```

Comment:
I think it might be clearer if we left this as an immutable one, since the domain of the origin can't change while the document is being fetch.

The one case I'm unsure about is if this can represent the origin of an about:blank loaded in a document that has set document.domain. If we need to provide a MutableOrigin for that, then I guess this needs to stay as-is.